core/src/sumeragi/main_loop.rs

```diff
@@ -463,123 +465,129 @@ fn handle_role_agnostic_messages<F>(
+                    if hash == voting_block_hash.transmute() {
+                        add_signatures::<true>(&mut voted_block, signatures.transmute());
 
-                commit_block(sumeragi, block, state);
+                        match voted_block.block.commit(&state.current_topology) {
+                            Ok(committed_block) => commit_block(sumeragi, state, committed_block),
+                            Err((_, err)) => {
+                                error!(%addr, %role, %hash, ?err, "Block failed to be committed")
+                            }
+                        };
+                    } else {
+                        error!(
+                            %addr, %role, committed_block_hash=%hash, %voting_block_hash,
+                            "The hash of the committed block does not match the hash of the block stored by the peer."
+                        );
+
+                        *voting_block = Some(voted_block);
+                    };
+                } else {
+                    error!(%addr, %role, %hash, "Peer missing voting block")
+                }
             }
-            other => *maybe_incoming_message = Some(other),
+            other => *incoming_message = Some(other),
         }
     }
 }
 
 #[allow(clippy::too_many_arguments)]
-fn compare_view_change_index_and_block_height_to_old<F>(
-    sumeragi: &SumeragiWithFault<F>,
+fn reset_state(
+    addr: &str,
     current_view_change_index: u64,
     old_view_change_index: &mut u64,
     current_latest_block_height: u64,
     old_latest_block_height: &mut u64,
     // below is the state that gets reset.
     current_topology: &mut Topology,
-    voting_block_option: &mut Option<VotingBlock>,
-    block_signature_acc: &mut Vec<SignatureOf<SignedBlock>>,
-    has_sent_transactions: &mut bool,
-    instant_when_we_should_create_a_block: &mut Instant,
-) where
-    F: FaultInjection,
-{
-    if current_latest_block_height != *old_latest_block_height {
-        *voting_block_option = None;
-        block_signature_acc.clear();
-        *has_sent_transactions = false;
-        *instant_when_we_should_create_a_block = Instant::now() + sumeragi.block_time;
+    voting_block: &mut Option<VotingBlock>,
+    voting_signatures: &mut Vec<SignatureOf<SignedBlock>>,
+    round_start_time: &mut Instant,
+    view_change_suggested: &mut bool,
+    view_change_proof_chain: &mut Vec<Proof>,
+) {
+    let mut was_commit_or_view_change = false;
 
+    if current_latest_block_height != *old_latest_block_height {
         *old_latest_block_height = current_latest_block_height;
+        was_commit_or_view_change = true;
     }
     if current_view_change_index != *old_view_change_index {
         current_topology.rebuild_with_new_view_change_count(current_view_change_index);
+        *old_view_change_index = current_view_change_index;
+        was_commit_or_view_change = true;
+        error!(%current_view_change_index, "No consensus, updating view change...");
```

Comment:
```suggestion
        warn!(%current_view_change_index, "No consensus, updating view change...");
```

`error` is reserved for things that the user is expected to fix. Given that BFT is a selling point, asking the user to investigate is not particularly useful.

Reply:
in this case your point is valid. However, due to the nature of distributed systems it's not possible to always say if error is in the system, user made or it's just a flake. Like when you get failed signature verification, this can be due to some misconfiguration, malicious peer, or just because some message came in late (which is normal behavior). These cases require manual inspection by a human and I lean toward marking them as `error`s in the log even if they may not be